MDANSE/Src/MDANSE/Framework/Jobs/XRayStaticStructureFactor.py

```diff
@@ -130,41 +140,85 @@ def finalize(self):
 
         dr = self.configuration["r_values"]["step"]
 
-        nAtomsPerElement = self.configuration["atom_selection"].get_natoms()
-        for pair in self._elementsPairs:
-            pair_str = "".join(map(str, pair))
-            if self.indices_intra is not None:
+        for label, _ in self.labels:
+            self._outputData.add(
+                f"xssf_{label}",
+                "LineOutputVariable",
+                (nq,),
+                axis="q",
+                units="au",
+                main_result=True,
+                partial_result=True,
+            )
+        self._outputData.add(
+            "xssf_total",
+            "LineOutputVariable",
+            (nq,),
+            axis="q",
+            units="au",
+            main_result=True,
+        )
+        if self.intra:
+            for label, _ in self.labels_intra:
                 self._outputData.add(
-                    f"xssf_intra_{pair_str}",
+                    f"xssf_intra_{label}",
                     "LineOutputVariable",
                     (nq,),
                     axis="q",
                     units="au",
                 )
+            for label, _ in self.labels:
                 self._outputData.add(
-                    f"xssf_inter_{pair_str}",
+                    f"xssf_inter_{label}",
                     "LineOutputVariable",
                     (nq,),
                     axis="q",
                     units="au",
                 )
             self._outputData.add(
-                f"xssf_{pair_str}",
+                "xssf_intra_total",
+                "LineOutputVariable",
+                (nq,),
+                axis="q",
+                units="au",
+            )
+            self._outputData.add(
+                "xssf_inter_total",
                 "LineOutputVariable",
                 (nq,),
                 axis="q",
                 units="au",
-                main_result=True,
-                partial_result=True,
             )
 
-            ni = nAtomsPerElement[pair[0]]
-            nj = nAtomsPerElement[pair[1]]
-
-            idi = self.selectedElements.index(pair[0])
-            idj = self.selectedElements.index(pair[1])
+        nAtomsPerElement = self.configuration["atom_selection"].get_natoms()
 
-            if pair[0] == pair[1]:
+        def calc_func(
+            label_i: str, label_j: str
+        ) -> Iterator[tuple[str, bool, np.ndarray]]:
+            """Calculates the xray static structure factor for a given
+            pair of element labels.
+
+            Parameters
+            ----------
+            label_i : str
+                The element label.
+            label_j : str
+                The element label.
+
+            Yields
+            ------
+            tuple[str, bool, np.ndarray]
+                A tuple of the results name, a bool specifying whether
+                results correspond to intermolecular atom pairs, and
+                the results.
```

Comment:
Numpydoc allows for:
```suggestion
            name : str
                Results name
            inter : bool
                Whether results are intermolecular atom pairs.
            results : np.ndarray
                The results.
```